sycl/doc/extensions/experimental/sycl_ext_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -0,0 +1,929 @@
+Nvidia Tensor Cores. This provides a portable and performant API for
+users who want to build their own neural networks applications,
+perform custom optimzations, or experiment with new operations in a
+timely and performing manner.
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification. An implementation supporting this extension must predefine
+the macro `SYCL_EXT_ONEAPI_MATRIX` to one of the values defined in the
+table below. Applications can test for the existence of this macro to
+determine if the implementation supports this feature, or applications
+can test the macro's value to determine which of the extension's
+features the implementation supports.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|The APIs of this experimental extension are not versioned, so the
+ feature-test macro always has this value.
+|===
+
+=== New `joint_matrix` class
+This extension adds a new class named `joint_matrix`, which represents
+a small 2-dimensional matrix that supports native operations in
+hardware. There are a number of template parameters, namely the group
+scope, the type of the elements, the matrix use, the shape, and the
+memory layout of the matrix.  This results in the following description:
+
+```c++
+namespace sycl::ext::oneapi::experimental::matrix {
+
+template <typename Group, typename T, use Use, size_t Rows, size_t Cols,
+          layout Layout = (Use == use::accumulator) ?
+                          layout::dynamic : /*unspecified*/ >
+struct joint_matrix {
+  joint_matrix();
+  joint_matrix(const joint_matrix &) = delete;
+  joint_matrix &operator=(const joint_matrix &) = delete;
+};
+
+} // namespace sycl::ext::oneapi::experimental::matrix
+```
+The constructor for the `joint_matrix` type is a group function as
+defined in section 4.17.3 of the core SYCL specification. It must be
+encountered in converged control flow by all work-items in the
+`Group`.
+
+==== Group Memory Scope
+Most operations on the joint_matrix are group functions, meaning that
+all work items in a group collectively perform an operation on the
+same matrix. The `Group` template parameter specifies the execution
+scope of the work-items in the group. The `joint_matrix` is shared among the
+work items in the group and is not private to each work item. This
+extension currently supports only the sub-group scope, so the `Group`
+template parameter must be `sycl::sub_group`, and group operations for
+the joint matrix must be done collectively by the work-items in a
+single sub-group. In this case, a matrix is declared as follows:
+
+```c++
+joint_matrix<sub_group, int8_t, use::a, tM, tN, layout::row_major> tA;
+```
+
+==== Element Type
+The `T` template parameter specifies the type of each element in the
+matrix. Each device supports only certain element types, so the
+application must use the query operations (defined below) to ensure
```

Comment:
```suggestion
application can use the query operations (defined below) to ensure
```

Reply:
I'm not sure about this comment.  Isn't "must" better?  The application is responsible for ensuring that the element type is supported on the device, otherwise the kernel won't run.

Reply:
I don't think it is very important, but this currently says any application using `joint_matrix` must query the parameters passed to `joint_matrix` using the query interface. 
This isn't strictly true since a `joint_matrix` will construct fine with correct params without checking the query interface, and indeed if a user already knows through some means that the params are OK for all backends they intend on running the application on then it will run fine. But this seems pretty obvious so I don't think the wording choice of must/can is super important.

Reply:
How about wording it like this then:

> The `T` template parameter specifies the type of each element in the matrix. Each device supports only certain element types, so the application must ensure that the element type is supported on the device where the kernel using this `joint_matrix` runs.  The query functions (defined below) may be used to determine the set of element types that are supported on a device.

Reply:
Sounds good!